APIDesafioWarren/Controllers/CustomerController.cs

```diff
@@ -40,206 +39,78 @@ public IActionResult GetById(int id)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                     .GetBy(x => x.Id.Equals(id));
 
-                return customer is null 
+                return customer is null
                     ? NotFound($"Customer not found! for id: {id}")
                     : Ok(customer);
             });
         }
 
-        [HttpGet("Byfullname/{fullname}")]
-        public IActionResult GetByfullname(string fullname)
+        [HttpGet("full-name/{fullname}")]
+        public IActionResult GetByFullName(string fullname)
         {
             return SafeAction(() =>
             {
-                var customer = _customerServices
+                var customer = _customerAppService
                     .GetAll(c => c.FullName == fullname);
 
-                return customer.Count is 0 
+                return customer.Count() is 0
                     ? NotFound("Customer not found!")
                     : Ok(customer);
             });
         }
 
         [HttpGet("Byemail/{email}")]
```

Comment:
```suggestion
        [HttpGet("email/{email}")]
```
O padrão é sempre deixar em minúsculo e não precisa do 'By'.